### PR TITLE
Prevent double loading at the same time, make api calls parallel, refetch everything when loading the page

### PR DESCRIPTION
**Pull request by MaikelStuivenberg**
## Description

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 34d0c1dfb77039983016171d5b230e786586fd39  | 
|--------|--------|

### Summary:
Enhanced the loading and fetching mechanisms for history data in the application, including preventing double loading, making API calls in parallel, and refetching data on page load.

**Key points**:
- Prevent double loading in `HistoryCubit` by checking load status before proceeding.
- Implement parallel API calls in `HistoryCubit` for fetching donation and allowance history.
- Refetch history data from the beginning when loading the `HomeScreen`.
- Minor adjustments in `routes.dart` and `home_screen.dart` to integrate changes.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by MaikelStuivenberg** [comment]
@ellipsis-dev review this

File: lib/features/family/features/history/history_cubit/history_cubit.dart [hunk 1]
```diff
@@ -12,22 +12,35 @@ class HistoryCubit extends Cubit<HistoryState> {
...
+      final allowanceHistoryFuture = historyRepo.fetchHistory(
         childId: childId,
         pageNumber: state.pageNr,
         type: HistoryTypes.allowance,
       );
 
+      // In this way the second future will already start fetching
+      // while waiting for the first one to finish
+      // This will make the fetching faster
+      final donationHistory = await donationHistoryFuture;
```
Comment (TammiLion) [inline lib/features/family/features/history/history_cubit/history_cubit.dart:41]: I don't think it does what you say it does...

Scherm­afbeelding 2024-06-12 om 09 51 52

If you want to fetch the calls in parallel I would recommend the Future.wait([]) function:
https://api.dart.dev/stable/3.4.3/dart-async/Future/wait.html

so:

`var responses = await Future.wait([donationHistoryFuture, allowanceHistoryFuture])`

**Review by TammiLion — APPROVED** [review]
Please take a look at the comment, for the rest it looks good.

File: lib/features/family/features/history/history_cubit/history_cubit.dart (hunk 1 above)
Comment (MaikelStuivenberg) [reply lib/features/family/features/history/history_cubit/history_cubit.dart:41]: You can have futures without awaiting them :) they also will get executed immediately.

Comment (MaikelStuivenberg) [reply lib/features/family/features/history/history_cubit/history_cubit.dart:41]: so the execution process will start already before the first await

Comment (MaikelStuivenberg) [reply lib/features/family/features/history/history_cubit/history_cubit.dart:41]: I think the future.wait is only needed if you need to be sure at that point that it has both results. But in our case its fine if it doesnt have the second one yet

Comment (TammiLion) [reply lib/features/family/features/history/history_cubit/history_cubit.dart:41]: Good point, indeed nothing needs to be changed then!

**Pull request merged**